ion_functions/data/generic_functions.py

```diff
@@ -515,3 +515,38 @@ def bilinear_interpolation(x, y, points):
 
 def error(x, y):
     return np.abs(x - y) / np.abs(y)
+
+
+def select_non_zero_arg(a1, a2=None, a1_scale_factor=None, a2_scale_factor=None):
+    """
+    Description:
+
+        Test arrays a1 and a2 for non-zero values and return the non-zero array.
```

Comment:
`Test arrays a1 and a2 for non-zero values and return the non-zero array, preferring a1 if both are present.`